### ISSUE TITLE
Reporting reasons are no longer being added as comments on MS

### ISSUE THREAD

**Issue by gparyani** [Charcoal-SE/SmokeDetector#3874]
Recently, #2398 was implemented, which automatically adds reporting reasons (i.e. `!!/report [post link] "reporting reason"`) as a comment on the relevant metasmoke record for the post.

However, this seems to have broken at some point in the last month or two.

Can this please be fixed?

**Comment by makyen** [comment]
Please provide an example of this not functioning as designed.

The most recent example from CHQ appears to be [this `!!/report`](https://chat.stackexchange.com/transcript/11540?m=54454342). The two MS reports ([1](https://m.erwaysoftware.com/posts/uid/stackoverflow/61962932), [2](https://m.erwaysoftware.com/posts/uid/stackoverflow/61962991)) created from the `!!/report` contain the reporting information in the `why` data, which is the way this has always worked since it was implemented.

**Comment by gparyani** [comment]
@makyen That's how it *used* to work prior to the implementation of the linked issue. After its implementation, those used to not only be added to the `why` data, but *also* as a comment posted on the MS record on behalf of the reporting user. It is not working as designed.

See these two posts showing how it was working after that issue was implemented:

- https://metasmoke.erwaysoftware.com/post/209777
- https://metasmoke.erwaysoftware.com/post/209983

Was it de-implemented at some point?

**Comment by makyen** [comment]
OK. My bad. It looks like it's supposed to be doing so, but is not.

**Comment by gparyani** [comment]
@user12986714 Unfortunately, it seems your above PR didn't fix the issue. I [tried using the manual report feature just now](https://metasmoke.erwaysoftware.com/post/243998) and it didn't post as a comment.

**Comment by gparyani** [comment]
Figured out how to [search](https://metasmoke.erwaysoftware.com/search?page=2&utf8=%E2%9C%93&why=with+reason) for manually reported posts with custom reasons specified.

Looking at that list, it seems that this change *never really worked* except for the two examples I gave in my prior comment here. I guess that's why you @makyen thought it was by design at first.

It seems that while that request was implemented in October 2019, it didn't work until those two reports in November 2019, and stopped working again since.
